### ISSUE TITLE
Web pages never load - Firefox Nightly, uBlock 1.18.4

### ISSUE THREAD

**Issue by sciguyryan** [uBlockOrigin/uBlock-issues#416]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [ ] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

Pages will not load in Firefox nightly (after 28/1/2019) with uBlock enabled. Pages will never load as soon as the addon is enabled. Disabling the addon instantly corrects the issue. This appears to function everywhere.

I had initially thought that Mozilla had somehow broken things, though after going through and disabling addons one at a time it appears to be a uBlock related issue.

The following can be found in the browser console: https://pastebin.com/GXbeDXnm

### A specific URL where the issue occurs

https://google.com/

### Steps to Reproduce

1. Enabled uBlock.
2. Refresh a web page. Pages will not load at all.
3. Disable uBlock (or revert the browser to the 28/1 build).
4. Web pages will once again load.

### Expected behavior:

Pages should load correctly with the addon enabled.

### Actual behavior:

Pages never finish loading with the addon enabled.

### Your environment

* uBlock Origin version: 1.18.4
* Browser Name and version: Firefox Nightly - 28/01/2019 build
* Operating System and version: WIndows 10, 64-bit.

**Comment by uBlock-user** [comment]
> The following can be found in the browser console: https://pastebin.com/GXbeDXnm

Your nightly appears to be a broken build. I'm on 67.0a1 (2019-02-02) (64-bit) and cannot reproduce.

**Comment by uBlock-user** [comment]
Update to the latest nightly and try to reproduce.

**Comment by sciguyryan** [comment]
> 
> 
> Update to the latest nightly and try to reproduce.

I did, hence step 3. I can resolve the issue by either downgrading nightly or by disabling uBlock.

**Comment by uBlock-user** [comment]
> Firefox Nightly - 28/01/2019 build

but your browser version says January build, this is why i said update to the latest nightly build

**Comment by uBlock-user** [comment]
Updated to the latest Nightly on my device, web pages load normally as expected, but I do one debug console error like you do.

![](https://i.gyazo.com/38117316f8bd6f749beb93d156171444.png)

**Comment by sciguyryan** [comment]
I'm going to give a clean profile a try and see if I can unpick what's going on then. It's particularly odd that it works perfectly if I revert back to the specific nightly build, though.

**Comment by uBlock-user** [comment]
Nightly is unstable and can be broken at any new build, so not a surprise.

**Comment by gorhill** [comment]
There is a `wasm` error in your console which make me think your indexedDB storage is corrupted. If you know were to find it, you could delete it manually and try restart uBO again.

**Comment by LinuxOnTheDesktop** [comment]
I have a somewhat similar problem; tell me whether I should open a new issue, please.

My problem is as follows and pertains to Firefox 65.0 (which is a 'release' build). When Firefox is open, and I click a link in Thunderbird, then, sometimes - depending on the URL? - and only when Ublock Origin is enabled for the page in question, the following occurs. Firefox will not load the page and indeed parts of the Firefox UI do not work. I have had this problem with URLs starting: https://bugzilla.mozilla.org/

**Comment by weaversam8** [comment]
I have also experienced this issue. Logged into my laptop today to find no web pages loading. I'm running Firefox Developer Edition, `66.0b6` build ID `20190207161357`, update channel `aurora`, on Windows 10. 

Disabling the add-on fixed the problem, and the problem persisted even after the add-on was uninstalled, reinstalled, and reset to default settings.

**Comment by gorhill** [comment]
Any errors reported at the browser console? (ctrl-shift-J)

**Comment by weaversam8** [comment]
Tried both in a fresh profile and in my existing profile and the issue has entirely disappeared. Unable to reproduce all of a sudden :/

**Comment by LinuxOnTheDesktop** [comment]
@weaversam8 

So in both your case and mine the problem suddenly disappeared. It does not seem as though there was some thing we both did, so I infer that the problem is intermittent, i.e. dependent on things we know not. But perhaps I miss something.

**Comment by gorhill** [comment]
There was an issue resembling this in bugzilla, <https://bugzilla.mozilla.org/show_bug.cgi?id=1496684>, but this was affecting Firefox for Android. I wonder if a similar, rare condition could also be occurring in Firefox Desktop.

Also, one thing I wonder is if the [code here](https://github.com/gorhill/uBlock/blob/74823f61209bf5d83ef9023c69ec41f894f2961c/platform/firefox/vapi-webrequest.js#L177) can be re-entrant -- I have assumed it is not. If there is re-entrance, this could cause some network requests to never be processed at start up. I strongly doubt re-entrance is possible, but nevertheless I will modify to be shielded against this in the next dev build.

**Comment by weaversam8** [comment]
I caught the issue yesterday evening on my other PC before it disappeared, and took a screenshot of the browser console. There's a lot of noise here, probably coming from other extensions, but here's one full page screenshot in case anything jumps out...

![image](https://user-images.githubusercontent.com/2546219/52647293-71cfa200-2eb2-11e9-9d2f-0a42de46167a.png)

**Comment by gorhill** [comment]
I can't find related to uBO in that console output. There seems to be another extension causing all these errors/warning. What else do you have installed?

**Comment by Sejsel** [comment]
Had the issue with the 65.0.1 (64-bit) stable build that was released yesterday.

After disabling "I am an advanced user" and purging all filter caches, then disabling and enabling the extension I no longer have the issue. I have re-enabled the advanced settings and everything seems to be working now.

When having this issue, I noticed there were no requests being made in the network tab of dev tools at all.

**Comment by uBlock-user** [comment]
Seems like it was a case of broken Nightly build..

**Comment by LinuxOnTheDesktop** [comment]
@uBlock-user I - and others, I believe - have the problem on regular ('release') Firefox.

**Comment by uBlock-user** [comment]
@LinuxOnTheDesktop Can you still reproduce this ?

**Comment by LinuxOnTheDesktop** [comment]
Yes, on UBO 1.18.4 on Linux.

**Comment by andyklimczak** [comment]
Experiencing this on Firefox Beta 66.0b4 (64-bit) on linux.
First pages will take forever to load. And will eventually load after a few minutes.
They instantly load if I go to `about:addons` and disable `uBlock`

**Comment by gorhill** [comment]
There is no point just saying you also have the issue, please also provide what is asked above: browser console error, new profile, other extensions interfering, etc. That the issue disappear when disabling uBO does not mean at all the issue lies with uBO, this needs to be furtehr investigated. Since I can't reproduce, someone else will need to step up to investigate beyond reporting "me too".

**Comment by uBlock-user** [comment]
Updated to the latest Nightly and still cannot reproduce, those who can reproduce post a different set of repro steps for further investigation, otherwise this issue will not go anywhere beyond "me too".

**Comment by LinuxOnTheDesktop** [comment]
Here are two sets of - somewhat overlapping - reports from the browser console. The output in report 1 (or some of it, anyway) was produced when, with Firefox (65.0, on Linux) open, I tried to open a Firefox tab, via Thunderbird, onto (so to speak) a github page. The output in report 2 was generated in the way way, including it being github that I was visiting, though this time the github page was this very page here. The second report was generated a few minutes are generating the first.

[Report 1 (pastebin)](https://pastebin.com/FKabyfk9)
[Report 2 (pastebin)](https://pastebin.com/iVRLV4Du)
Here's some [details of my Firefox configuration, too (also on Pastebin)](https://pastebin.com/feycnujf)

**Comment by gorhill** [comment]
I see nothing from uBO in there. I tried opening the current page from within a Thunderbird link, and it opened fine. Does it happen every time you try or just once in a while?

**Comment by LinuxOnTheDesktop** [comment]
At present the problem seems constant, for gitbhub and bugzilla (so: bugs when reporting bugs - _deeply_ irritating) but the problems with those sites have proved to be merely intermittent in the past. Perhaps the people having problems have some addOns, aside from Unlock, in common. Or else some Firefox setting perhaps.

**Comment by LinuxOnTheDesktop** [comment]
I think the problem may have to do, at least on Linux, with Firefox being set to show tabs in the titlebar.

**Comment by gorhill** [comment]
It's why people are asked to find out if an issue also occurs in a new profile with only uBO, this reduces the amount of external variables when trying to narrow down an issue.

**Comment by LinuxOnTheDesktop** [comment]
@gorhill 

Sorry. I have now tested with a new profile and with that I am _not_ - at present anyway - having the problem, even when I tell Firefox to draw titlebars in tabs.

**Comment by LinuxOnTheDesktop** [comment]
The problem has reoccured; it is intermittent. Here is some more data from the console (existing profile, Linux) when the bug occured. Hopefully this report contains something relevant (though I fear it does not).
[ConsoleOutput.txt](https://github.com/uBlockOrigin/uBlock-issues/files/2892281/ConsoleOutput.txt)

**Comment by LinuxOnTheDesktop** [comment]
Eventually Firefox loaded some pages. New log attached. (The log includes . . a Reddit logo in
[new.txt](https://github.com/uBlockOrigin/uBlock-issues/files/2892301/new.txt)
 ASCII art . .?)

**Comment by gorhill** [comment]
There is nothing wrong in there, it appears stuff was blocked as expected and it's reported in the console.

**Comment by uBlock-user** [comment]
Does this happen on cold start/first start of the browser ?

**Comment by LinuxOnTheDesktop** [comment]
> Does this happen on cold start/first start of the browser?

In my case, I _think_ Firefox has to open already.

**Comment by dont-bug-me** [comment]
I'm not 100% sure if this is the same issue, but if i go to [this ](https://www.maniac.de)site it stays completely blank. But I found out what caused the problem at least on that site. The problem has to do with cosmetic filtering. In my uBo settings i have "Parse and enforce cosmetic filters" activated and "Ignore generic cosmetic filters" deactivated. When I deactivate "Parse and enforce cosmetic filters" or activate "Ignore generic cosmetic filters" the site loads just as expected.
I can reproduce this behavior with uBo 1.18.4 on the latest stable Firefox and on Firefox 66.0b11.

**Comment by gorhill** [comment]
@dont-bug-me Filter issue, report the issue to EasyList Germany please.

**Comment by LinuxOnTheDesktop** [comment]
On Linux, I have "Ignore generic cosmetic filters" enabled and I still have the problem.

**Comment by quassy** [comment]
I can confirm this with uBlock Origin 1.18.4 on Firefox 66.0b13 (64-Bit) on Windows 10 with an existing profile (with my normal extensions as well as uBlock Origin being the only extension) and had this problem since monday, but I am too unable to reproduce with a new profile. What might give a hint for troubleshooting is that, after restarting Firefox, uBlock does not remember any settings (e.g. Show number of blocked elements on icon) or custom filters, even the [dynamic filtering](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-rule-syntax) list and [white list](https://github.com/gorhill/uBlock/wiki/Dashboard:-Whitelist) are completely blank, even though they should be filled by default.

This does affect not most of my extensions, but Tab Center Redux is also broken and does not remember any settings.

When debugging uBlock Origin and trying to change some settings (checking checkboxes, adding a new custom filter), the same types of error messages are logged in console on every operation (the output for Tab Center Redux is similar, always referring to ExtensionUtils.jsm and storage.js:633):

```
Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionUtils.jsm
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:633
Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionUtils.jsm
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:633
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. assets.js:682
TypeError: "Response has unsupported MIME type" lz4-block-codec-wasm.js:148:17
Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionUtils.jsm
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:633
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. assets.js:682
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. assets.js:701
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:633
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. assets.js:682
Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionUtils.jsm
```

**Comment by uBlock-user** [comment]
>  What might give a hint for troubleshooting is that, after restarting Firefox, uBlock does not remember any settings (e.g. Show number of blocked elements on icon) or custom filters, even the dynamic filtering list and white list are completely blank, even though they should be filled by default.

This does affect not most of my extensions, but Tab Center Redux is also broken and does not remember any settings.

Different [issue](https://github.com/gorhill/uBlock/issues/3071) suffered by using private mode/never remembering history, not related to this. Fixed in the dev build.

**Comment by quassy** [comment]
To add to my comment: The errors seem to be related to IndexedDB, which is probably the way uBlock Origin and Tab Center Redux save their settings. Deleting the contents of `profile_folder/storage/` solves the problem, so maybe the databases there got corrupted by a Firefox update.

> Different issue suffered by using private mode/never remembering history, not related to this. Fixed in the dev build.

I am not using always-private mode and do keep my browsing history, only blocking third-party cookies and always use tracking protection (list 1).

**Comment by gorhill** [comment]
Form the errors you report above, it seems the issue is with Firefox WebExtensions's `browser.storage.local` -- so unrelated to uBO's use of IndexedDB for caching purpose. This could be a case of corrupted storage.

**Comment by gorhill** [comment]
@uBlock-user 

> Fixed in the dev build.

The reported error messages are unrelated to this. The issue as per error messages is with `browser.storage.local`. This might be related to the fact that Firefox migrated extension storage from JSON-based to IndexedDB-based.

**Comment by gorhill** [comment]
@quassy Do you have a lot of custom filters? I am wondering if Firefox started to enforce `unlimitedStorage` permission.

**Comment by quassy** [comment]
> @quassy Do you have a lot of custom filters? I am wondering if Firefox started to enforce `unlimitedStorage` permission.

Not really, but can't tell for sure (as I have reset storage now and restoring it will restore the bug / make the list impossible to view). Probably less than 100, so nowhere near any storage limit Firefox will enforce ([I'd guess something between 10 MB to 2 GB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Browser_storage_limits_and_eviction_criteria#Storage_limits)). Also Tab Center Redux has only minimal settings and is even less likely to hit any limit. 

Is there any way to find out, which storage files belong to uBlock, to check its size (and maybe contents with SQLite)?

**Comment by gorhill** [comment]
> Is there any way to find out, which storage files belong to uBlock

Here is what I observe with my Nightly/Linux installation:

- `[path-to-profile]/browser-extension-data/uBlock0@raymondhill.net/storage.js.migrated`
- `[path-to-profile]/storage/default/moz-extension+++[uBO-unique-id]^userContextId=4294967295/idb/3647222921wleabcEoxlt-eengsairo.sqlite`

You can find `uBO-unique-id` in `about:debugging#addons`. The numeric values seen in there might be different on your side I suppose.

The sizes on my side are as expected, under 100KB.

**Comment by agentjsp** [comment]
I get this same issue with the Firefox 66 release candidate. Happened right after the last update (like a day ago) and didn't happen on the last beta build. I'm on the beta channel.

I have uBlock Origin disabled until further notice, because no page loads when it's enabled.

**Comment by gorhill** [comment]
Your cache storage might be corrupted, force a hard purge with ctrl-shift-click "Purge all caches".

**Comment by gorhill** [comment]
Try uBO in a new profile with no other extensions or `about:config` changes. As often mentioned, something else might interfere negatively and uBO is merely making this apparent.

**Comment by uBlock-user** [comment]
Should this still be kept open given it's going nowhere and OP has failed to provide feedback ?

**Comment by agentjsp** [comment]
@gorhill It appears to work in a new profile. Absolutely nothing has changed between the update though (besides going from the latest beta to RC1), so if something is interfering negatively the add-on may need to be updated or the Firefox devs need to be notified. I cleared my cache through the normal means, but I'm not sure if that clears the kind of storage mentioned in earlier posts.

@uBlock-user Why would it be closed given that it's a critical bug that is preventing some people from using the software?

**Comment by uBlock-user** [comment]
> Why would it be closed given that it's a critical bug 

because it's still not proven to be a uBO bug, browser bugs which is likely this is, are dealt by browser devs.

**Comment by agentjsp** [comment]
My understanding is the fact that it hasn't yet been proven to not be related to uBO is reason for the issue to remain open, so that feedback can continue to be gathered and troubleshooting can be consolidated into a single issue. It didn't occur on my end until yesterday, so at the minimum we know that it is still happening.

**Comment by gorhill** [comment]
> the add-on may need to be updated or the Firefox devs may need to be notified

Not necessarily. Maybe this is a consequence of your on-disk storage being corrupted. Browser or OS crash, or killing the browser from Task Manager can cause such data corruption.

Also, it could be some settings you have changed in `about:config`, or another extension misbehaving preventing uBO to get CPU time (all extensions execute in the same process).

It's why I always aks to try uBO in a new profile with no changes. Then once it's confirmed it works fine, then you will have to further investigate to narrow the cause of the issue.

**Comment by gorhill** [comment]
By the way, one thing you can try is set `suspendTabsUntilReady` to `no` [in advanced settings](https://github.com/gorhill/uBlock/wiki/Advanced-settings#suspendtabsuntilready-experimental). See if this changes anything. If this seems to help, it's most likely just a remedy for the symptom, not the root cause.

**Comment by agentjsp** [comment]
I tried it with all other add-ons disabled, and there are no about:config settings I can think of.

It also worked with the exact same profile prior to the upgrade, so perhaps the installer does something to existing profiles that triggers a compatibility and/or somehow corrupted data. I haven't noticed any issues with other add-ons though. Is there a way I can reset the on-disk storage you're referring to?

**Comment by agentjsp** [comment]
One thing I just noticed is that support.mozillla.org loads, but I can't seem to find any other domains that will. Nothing else loads including mozilla.org.

> By the way, one thing you can try is set `suspendTabsUntilReady` to `no`

This does fix it after a browser restart, though I notice that uBO says `0` for every single page. When putting this back to `unset` and restarting the problem resumes.

**Comment by uBlock-user** [comment]
> uBO says 0 for every single page

As in 0 or 0 % request blocked ?

**Comment by gwarser** [comment]
> I just noticed is that support.mozillla.org loads

Privileged page: https://github.com/gorhill/uBlock/wiki/Privileged-Pages

**Comment by gorhill** [comment]
> though I notice that uBO says 0 for every single page

Is uBO still blocking properly? For example, do you see ads at this page: <https://www.reuters.com/article/us-california-drought-idUSKCN1QW09A>? (I do only if I disable uBO on that site).

**Comment by agentjsp** [comment]
On Forbes.com it says 0 under "This Page", and curiously the advertisements are all blocked. When I try Forbes.com on another profile it says 9.

For https://www.reuters.com/article/us-california-drought-idUSKCN1QW09A the ads are not blocked, though it still says 0 while the other profile says 15. The filters are the same.

**Comment by gorhill** [comment]
> Is there a way I can reset the on-disk storage you're referring to?

Ctrl-Shift-clicking the _"Purge all caches"_ button should outright delete uBO's indexedDB used for asset caching purpose. See if this makes a difference.

> On Forbes.com it says 0 under "This Page", and curiously the advertisements are all blocked

This is a curious symptom. Not sure what to make of this. Are there any error reported in the browser console? (Ctrl-Shift-j)

**Comment by agentjsp** [comment]
On Forbes.com I don't get any errors. I thought maybe it was Firefox content blocking that is blocking the ads, but that doesn't seem to be the case.

On the Reuters.com article I get the same warnings, as well as [these](https://i.imgur.com/jCWaA5s.png) errors. I'm only learning Japanese, but the second error says something about executing `postMessage` on `DOMWindow` failing, and complains about the destination address of the generating element.

**Comment by gorhill** [comment]
I see nothing related to uBO in these 3 messages. Just to be sure, this is the *browser* console? (not the page one's).

**Comment by agentjsp** [comment]
Whoops, that was the page one. I get [these](https://i.imgur.com/ZqZQQ1u.png) in the browser console. The one in Japanese is about text encoding.

**Comment by gorhill** [comment]
I see uBO-related stuff in there.

Can you clear the console, then disable uBO, wait a few seconds, then re-enable uBO and see what the console says?

**Comment by agentjsp** [comment]
I get [these](https://i.imgur.com/7GJ5end.png). Pretty sure the first two happened when it was disabled, and the rest after.

**Comment by gorhill** [comment]
I also see Firefox extensions framework failures in your screenshot, around:
https://searchfox.org/mozilla-central/rev/f1c7ba91fad60bfea184006f3728dd6ac48c8e56/toolkit/modules/addons/WebNavigationContent.js#215

**Comment by gorhill** [comment]
Can you expand the stack for the line `[uBlock0CacheStorage]`?

**Comment by gorhill** [comment]
I suspect a corrupted on-disk cache storage. However, I would expect uBO to just handle the issue and resume without causing initialization failure. What would be great is that you give me your cache storage DB for me to use as a test case to find out what occurs internally when uBO launches.

If you agree, these are the steps that come to mind as I write this:
- Find uBO's id assigned by Firefox:
    - Open `about:debugging`
    - Find uBO entry
    - Note the "Internal UUID"
- Find uBO's cache storage.
    - On my Linux installation, it's located at:
      `~/.mozilla/firefox/[Profile]/storage/default/moz-extension+++[Internal UUID]/`
        - Be sure to not confuse it with another closely name storage which ends with `userContextId=...`, I suspect this is the new back end for `browser.storage.local`.
    - Zip all the content of above folder and... maybe use Firefox Send?

**Comment by agentjsp** [comment]
It shows `genericErrorHandler moz-extension://f1560d25-6ade-4c86-8262-17cf32bb5792/js/cachestorage.js:152` when expanded.

I don't think I can send you the cache as I just tried the ctrl-shift-click on the "Purge all caches" button and it seems to have fixed the loading issue when `suspendTabsUntilReady` is returned to the default value. I'm certain the browser didn't crash, so I'm not sure what caused the original issue, or how/if it is related to this. When it's set to `no` Forbes.com still shows 0 while blocking ads and displays the same errors.

**Comment by agentjsp** [comment]
The loading issue is back again, and I'm certain there hasn't been a crash since I last posted.

**Comment by gorhill** [comment]
As per errors reported, an unknown error is occurring when uBO tries to access its own indexedDB. Do you have an anti-virus? Could your storage have become unreliable?

In any case, what you reported made me realize I need to change the use of `genericErrorHandler`, as it's not doing what it's supposed to do -- i.e. uBO may never end up resolving the Promise if ever that handler is called.

**Comment by uBlock-user** [comment]
is `dom.indexedDB.enabled` set to **false*` ?

**Comment by uBlock-user** [comment]
is `dom.indexedDB.enabled` set to **false** ?

**Comment by gorhill** [comment]
I am tempted to borrow [Firefox's implementation of extension storage using IndexedDB](https://hg.mozilla.org/integration/mozilla-inbound/file/tip/toolkit/components/extensions/ExtensionStorageIDB.jsm), it's quite simpler than the current implementation in uBO. I am still on a learning curve for ES6 Promise stuff, I can learn from Firefox implementation, they are using `async`/`await` and this seems to simplify a lot the usage of IndexedDB. Specifically, I don't see the code using success/error handlers, which I understand ensure that all errors are guaranteed to be handled through the Promises.

I can't fix this for the next release, but meanwhile those suffering those unexpected error issues can set `cacheStorageAPI` to `browser.storage/local` in advanced settings -- this will force uBO to use extension storage rather than a separate indexedDB-based one.

**Comment by gorhill** [comment]
> is `dom.indexedDB.enabled` set to **false** ?

It does not matter, uBO handles this properly -- it automatically falls back to extension storage. The issue here seems to be an unexpected error (`UnknownError`) which is not handled by uBO.

**Comment by gorhill** [comment]
I added a timeout for when uBO tries to open its indexedDB storage, currently set at 5s. See if this helps remove the hung page loads at launch time.

**Comment by agentjsp** [comment]
>As per errors reported, an unknown error is occurring when uBO tries to access its own indexedDB. Do you have an anti-virus? Could your storage have become unreliable? Do you have enough free space on it?

I only use Windows Defender which is built into the operating system. It's a SSD with 32GB free space, and I don't believe it is failing.

I installed 1.18.9rc2, but pages still load infinitely. I'm not sure if this is normal or not, but the SQLite file in the IDB folder is 20MB.

**Comment by gorhill** [comment]
Being able to reproduce on my side is the only way I can see moving forward with this. If it still infinitely load with 1.18.9rc2 then it means it's not failing when trying to access the DB. How is the CPU usage when it hangs?

**Comment by agentjsp** [comment]
I get [these](https://i.imgur.com/1WLrw8m.png) errors when starting up 1.18.9rc2. According to about:performance it appears to not be using any of the CPU, though I'm not sure how accurate that is. Firefox overall spikes to 10-20% when trying to load a new tab, though nothing ever actually starts loading.

I can still "fix" the infinite loading by either ctrl+shift clearing the cache (I tried this once yesterday, and it came back later) or setting `suspendTabsUntilReady` to no. Though I haven't tried either of those on 1.18.9rc2.

**Comment by grssam** [comment]
I am facing most of the issues mentioned here (tabs infinitely loading, IndexedDB related exception in browser console, browser works fine if addon disabled etc).
There is some code change related to Storage in Firefox 66 that is braking multiple addons in a way that even the browser/tabs fail to load/work properly if any such addon is enabled.

The way to reproduce is to have the addon already installed pre-Firefox 66 (say, fx65) and then upgrade to firefox 66.
The addon will work fine in a new/clean profile.
Since Firefox 66 rolled out today, I am assuming a LOT of users to face similar issue. Maybe the firefox devs will see as to why these issues are coming, maybe not.

**Comment by grssam** [comment]
Actually, I have tried completely uninstalling the addon and installing it fresh without any backup or restore... It still brakes my browser. After installing back, things would work fine, but any restart of browser leads to the exact same situation of infinitely loading tabs. While these tabs are loading, I simply disable uBO and tabs load instantly.

**Comment by LinuxOnTheDesktop** [comment]
If publicity is wanted for this problem, there's a rather fresh thread on Reddit's /r/linux.

Also, though: I thought the bug had struck when I tried to load this very Reddit page, but - and I've just updated Firefox from 65.0.1 (I think) to 66 - the page did, eventually, load. This is hardly a good test but if is representative then the bug has been mitigated somewhat (in that it has gone from tab-won't-load to tab-loads-slowly). Still: sciguyryan says, above, that moving to version 66 can _cause_ the problem.

**Comment by gorhill** [comment]
Whoever has the issue, set `cacheStorageAPI` to `browser.storage.local` in advanced setting, and see if that fixes the issue for you. Doing so will cause uBO to depends only on one single storage for both settings and cache.

**Comment by LinuxOnTheDesktop** [comment]
@gorhill : that's *Ublock's* advanced settings, right (as against Firefox's)? And those settings are the ones that one reaches by ticking, within the AddOn's options, 'I am an advanced user' and then clicking the cog? If so, then I take it that one has to enter 'cacheStorageAPI browser.storage.local' into the resulting box?

**Comment by gorhill** [comment]
Yes, replace

    cacheStorageAPI unset

With

    cacheStorageAPI browser.storage.local

Without touching anything else. Don't forget to apply changes.

**Comment by LinuxOnTheDesktop** [comment]
@gorhill 

Thank you that works: I have had no problems since doing that. Fingers crossed, but it's looking good! Github pages do load, when loaded via external applications (well, Thunderbird at least) rather slowly, but that might not be at all related to the problem at issue in this thread/bug report and at any rate the pages _do_ load. It would be good, of course, with Ublock and/or Firefox to ensure than no such hackery is needed in the future.

**Comment by gwarser** [comment]
Check for `browser.tabs.remote.autostart` in `about:config`. Should be `true`.

https://old.reddit.com/r/firefox/comments/b3lnw4/help_firefox_not_loading_pages_after_update_66/

**Comment by gorhill** [comment]
Well that is interesting -- was this the problem all along?

Whoever had the issue of hung tabs at launch, please verify your value for `browser.tabs.remote.autostart` and if it was set to `false`, please report here this was your issue, so as to get a sense of how much this was the actual issue here.

**Comment by LinuxOnTheDesktop** [comment]
I had tabs hanging when Firefox was launched by an external application and when Firefox was open already but a tab was opened by an external application. I had those problems on a Windows computer and a Linux computer. Creating the `cacheStorageAPI` key within Ublock and setting the content of the setting to `browser.storage.local` seems to have cured the problem on both computers. On both computers, Firefox's `browser.tabs.remote.autostart` was already `true`.

**Comment by agentjsp** [comment]
My `browser.tabs.remote.autostart` has been true and I still got/get the issue.

The catchStorageAPI tweak seems to work though.

**Comment by gorhill** [comment]
> The `catchStorageAPI` tweak seems to work though

You mean the one i just committed?

**Comment by agentjsp** [comment]
I'm just referring to setting it to `browser.storage.local` in the advanced settings.

**Comment by gorhill** [comment]
I would appreciate some volunteers try 1.18.11b0 after setting `cacheStorageAPI` back to `unset` to see if the issue still exists.

**Comment by LinuxOnTheDesktop** [comment]
Sure, I'll try it. I can test on both Windows and Linux. Got a link or some such for the extension?

**Comment by gorhill** [comment]
The dev build are always published in the [Releases section](https://github.com/gorhill/uBlock/releases).

**Comment by LinuxOnTheDesktop** [comment]
Here are some results with the beta. In all the scenarios that I describe `browser.tabs.remote.autostart` is `true`.


**Windows, with Firefox 66.0**

Loading this very webpage (viz., https://github.com/uBlockOrigin/uBlock-issues/issues/416):

W1) when the browser was not previously open, *works*;
W2) when the browser was open already, *works*.



**Linux, with Firefox _65.0.1_ (version 66 hasn't arrived on my distro - Mint - yet)**

Loading this very webpage:

L1) when the browser was not previously open, *works*;
L2) when the browser was open already, *hangs*,

I find further

L2') that 2 happens even when `catchStorageAPI` is set to `browser.storage.local`

and

L2'') 2 happens even when `catchStorageAPI` is set to `browser.storage.local` and when I use the *release* - not beta - version of the AddOn.

**Comment by LinuxOnTheDesktop** [comment]
Update: **Linux, with Firefox 66 (version 66 just arrived), with `catchStorageAPI` unset**

* when the browser was not previously open, works;
* when the browser was open already, _works_.

**Comment by gorhill** [comment]
When you say "works" on FF v66, this is with 1.18.11b0 only? Or is uBO's version unimportant?

**Comment by LinuxOnTheDesktop** [comment]
Upon further testing, I find that there is no problem with FF 66, on Linux, with _either_ the beta version of UBO _or_ with the release version of UBO, and without having to set the advanced options switch. So, in sum, it seems that Firefox 66, on its own, has solved the problem on both Platforms (Linux and Windows).

**Comment by ctjhoa** [comment]
FYI, I also got the same exact issue (infinite loading on any page).
Here is my configuration impacted by the issue:
- OS: Archlinux
- Firefox: 66.0.1 (64-bit)
- Firefox settings: `browser.tabs.remote.autostart: true`
- Ublock origin: `1.18.6`
- Ublock advanced settings: `cacheStorageAPI: unset`

Modifying Ublock advanced settings `cacheStorageAPI` from `unset` to `browser.storage.local` fix the issue.

I did some tests on other configurations I have.

Configurations which work perfectly fine:
My windows:
- OS: Windows 10
- Firefox: 66.0.1 (64-bit)
- Firefox settings: `browser.tabs.remote.autostart: true`
- Ublock origin: `1.18.6`
- Ublock advanced settings: `cacheStorageAPI: unset`

and my other arch:
- OS: Archlinux 
- Firefox: 65.0.2 (64-bit)
- Firefox settings: `browser.tabs.remote.autostart: true`
- Ublock origin: `1.18.6`
- Ublock advanced settings: `cacheStorageAPI: unset`

My filters & basic settings are the same across all my configurations.

**Comment by GerhardGitHub** [comment]
We have the same issue since update of Firefox to Version 66.0
in combination with browser.tabs.remote.autostart: **false**

I made the experience that the Problem only appears when starting FF 66.0 with Block active.

If FF ist running and Ublock is deactivated und instantly reactivated again in AddOns Configuration (not the uBlock-GUI), all pages load immediatly und the issue is gone until the next start of FF
with Ublock active.
Some behavior with different UBlock Verions and unrelated to cacheStorageAPI: unset
or otherwise.

My guess is, the problem is related to the initializing procedure of uBlock and FF 66.0.
Additional odd behavior is, that in some cases (not reproducable), the filter lists are
outdated even if they had been reloaded some short time before and wont update atuomatically .

**Comment by GerhardGitHub** [comment]
Just got FF 66.01
no issue even with FF-Setting browser.tabs.remote.autostart: false
and uBlock-Setting cacheStorageAPI: unset

**Comment by uBlock-user** [comment]
> browser.tabs.remote.autostart: false

that breaks new tab page on FF 68, beware of that

**Comment by agentjsp** [comment]
The add-on seems to work for me without any of the workarounds. I'm on 67 Beta 4 though. I'm guessing the Firefox devs did something in 66.0.1/67 beta after the release of 66 that fixed it? 

Apparently @ctjhoa got the issue even on 66.0.1 though, so I'm not sure.

**Comment by ixcat** [comment]
ff 66.0.1 osx
had the issue; browser.tabs.remote.autostart->true fixes the 'broken tab restart' issue;
did uninstall/reinstall ublock test without (ever) adjusting cachestorage api.

**Comment by vtpoet** [comment]
**Desktop:** Cinnamon 4.0.10 **Distro:** Linux Mint 19.1 Tessa
**firefox:** Installed: 66.0.1+linuxmint1+tessa

Just reporting if it's helpful. I've also been having this issue. 

**Firefox settings:** browser.tabs.remote.autostart: true
**uBlock Origin** v1.18.12
**Ublock advanced settings:** cacheStorageAPI: unset

The only solution that seems to work on my system is to turn off ublock, from the addons page, and turn it back on.

**Comment by gorhill** [comment]
> Firefox settings: browser.tabs.remote.autostart: true

What is your value for `extensions.webextensions.remote`?

**Comment by vtpoet** [comment]
Hi. The value is set to true.

**Comment by gorhill** [comment]
@vtpoet Did you restore a backed file with uBO 1.18.12?

**Comment by vtpoet** [comment]
No I didn't. Not smart enough to do that sort of JuJitsu. I saw that others had had success stopping and restarting, and tried that. Prior to that, I did disable all other addons, adding them back one by one, but none of that helped with Ublock.

**Comment by gwarser** [comment]
https://old.reddit.com/r/firefox/comments/b6yyx4/caution_when_using_latest_bleachbit_22_release/

>Caution when using latest Bleachbit (2.2) release

>I was unable to connect to any website using Firefox following a cleaning with the latest bleachbit update...

**Comment by xuhcc** [comment]
I have similar problem with uBlock. On my machine it always occurs on a first start of Firefox. Browser freezes for 2-3 minutes and only after that all tabs start loading. If I close browser and then reopen, it loads quickly. After reboot it freezes again.

**OS**: ArchLinux
**Firefox**: 66.0.2
**uBlock**: 1.18.14
**cacheStorageAPI workaround**: didn't help

At previous FF version (65) I had similar problem with another extension, IPFS companion (here's my bug report: ipfs-shipyard/ipfs-companion#655). I tried to find out what part of code caused freeze and it seemed to be `browser.webRequest.onBeforeRequest.addListener(onBeforeRequest, { urls: ['<all_urls>'] }, ['blocking'])`, but at this point of investigation freeze started to occur intermittedly so I gave up. After updating to FF66 something changed and now uBlock causes exactly the same freeze on a first start while IPFS companion works fine. All of it suggests that at least in my case the problem is with Firefox itself.

**Comment by doxxx** [comment]
Pages are never loading for me with uBlock Origin 1.18.14 on Firefox 67.0b6 on Windows 10 Enterprise 1803.

* The problem disappears when I disable the addon.
* The problem does _not_ go away when I simply disable uBlock for a site.
* The problem occurs when only uBlock Origin is enabled.

I turned on debugging for the addon and found the following in the addon's console log:\

```
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. background.js:18709
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:100
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:261
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. background.js:18709
TypeError: Argument 1 of PrecompiledScript.executeInGlobal is not an object. ExtensionContent.jsm:490:25
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. background.js:18709
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:100
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:261
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. background.js:18709
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:100
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. storage.js:261
Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionUtils.jsm
DOMException: "The operation failed for reasons unrelated to the database itself and not covered by any other error code." cachestorage.js:230:29
Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionUtils.jsm
TypeError: The expression cannot be converted to return the specified type.
Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. background.js:18709
```

I tried setting the `cacheStorageAPI` setting to `browser.storage.local` but it did not fix the problem.

**Comment by gorhill** [comment]
This issue has become pointless, it's now just a melting pot of all sort of possibly unrelated issues. Many of these where caused by `remote.autostart` set to `false`. The others might be cause by specific configuration which would explain why the issue can't be reproduce by most, including me. The issue should be close and whoever has a specific issue should open one with **all the steps** asked in the template being checked -- otherwise the issue will be closed. If it can't be reproduced by me and other volunteers, the assumption is that there is something specific on the user side which only hm can investigate.

**Comment by gorhill** [comment]
> I turned on debugging for the addon and found the following in the addon's console log:
> 
>     Unchecked lastError value: Error: UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. background.js:18709

There is no line 18709 in uBO's `background.js`, so such console log entry this makes no sense to me.

**Comment by doxxx** [comment]
Yeah, I noticed that. My suspicion is that it's something in Firefox 67.0b6 because everything was working on Friday, then I accepted the pending Firefox update this morning and everything was broken.

I'll post a new issue once I've tried it with a new profile.

**Comment by doxxx** [comment]
So after I figured out how to create a new profile, I was able to verify that the problem _does not occur_ in a new profile after installing uBlock. So something in my main profile is busted. I tried the [Reset Firefox feature](https://support.mozilla.org/en-US/kb/refresh-firefox-reset-add-ons-and-settings) and after reinstalling uBlock, I am _unable to reproduce the problem_.

So, TLDR: [Reset your Firefox profile](https://support.mozilla.org/en-US/kb/refresh-firefox-reset-add-ons-and-settings).

**Comment by gwarser** [comment]
One more edge case: foreign files in storage directory (but not standard system files, they are handled now by Fx) will cause endless loading until uBO is closed. 

However, this line will be shown in browser console:

>Quota Something (Text File) in the directory that doesn't belong!: ActorsParent.cpp:3905

**Comment by gorhill** [comment]
> foreign files in storage directory

But is it something uBO can and/or has to handle? The failure seems to be Firefox-side.
